zipline/assets/asset_writer.py

```diff
@@ -216,171 +232,174 @@ def write_version_info(version_table, version_value):
+            # Write the data to SQL.
+            self._write_df_to_table(
+                metadata.tables['futures_exchanges'],
+                data.exchanges,
+                txn,
+                chunk_size,
+            )
+            self._write_df_to_table(
+                metadata.tables['futures_root_symbols'],
+                data.root_symbols,
+                txn,
+                chunk_size,
+            )
+            asset_router = metadata.tables['asset_router']
+            self._write_assets(
+                asset_router,
+                metadata.tables['futures_contracts'],
+                'future',
+                data.futures,
+                txn,
+                chunk_size,
+            )
+            self._write_assets(
+                asset_router,
+                metadata.tables['equities'],
+                'equity',
+                data.equities,
+                txn,
+                chunk_size,
+            )
+
+    def _write_df_to_table(self, tbl, df, txn, chunk_size):
         df.to_sql(
             tbl.name,
-            bind.connection,
-            index_label=[col.name for col in tbl.primary_key.columns][0],
+            txn.connection,
+            index_label=first(tbl.primary_key.columns).name,
             if_exists='append',
-            chunksize=self.CHUNK_SIZE,
+            chunksize=chunk_size,
         )
 
-    def _write_assets(self, assets, asset_tbl, asset_type, bind):
-        self._write_df_to_table(assets, asset_tbl, bind)
-
-        pd.DataFrame({self.asset_router.c.sid.name: assets.index.values,
-                      self.asset_router.c.asset_type.name: asset_type}).to_sql(
-            self.asset_router.name,
-            bind.connection,
+    def _write_assets(self,
+                      asset_router,
+                      tbl,
+                      asset_type,
+                      assets,
+                      txn,
+                      chunk_size):
+        self._write_df_to_table(tbl, assets, txn, chunk_size)
+
+        pd.DataFrame({
+            asset_router.c.sid.name: assets.index.values,
+            asset_router.c.asset_type.name: asset_type,
+        }).to_sql(
+            asset_router.name,
+            txn.connection,
             if_exists='append',
             index=False,
-            chunksize=self.CHUNK_SIZE,
+            chunksize=chunk_size
         )
 
-    def _write_exchanges(self, exchanges, bind):
-        self._write_df_to_table(exchanges, self.futures_exchanges, bind)
-
-    def _write_root_symbols(self, root_symbols, bind):
-        self._write_df_to_table(root_symbols, self.futures_root_symbols, bind)
-
-    def _write_futures(self, futures, bind):
-        self._write_assets(futures, self.futures_contracts, 'future', bind)
-
-    def _write_equities(self, equities, bind):
-        self._write_assets(equities, self.equities, 'equity', bind)
-
-    def check_for_tables(self, engine):
+    def _check_for_tables(self, txn):
```

Comment:
Can we call this `has_all_tables` or something like it?  `check_for_tables` doesn't tell me what is being checked.